pkgs/tools/networking/xray/default.nix

```diff
@@ -0,0 +1,51 @@
+{ lib, fetchFromGitHub, fetchurl, symlinkJoin, buildGoModule, runCommand, makeWrapper, nixosTests
+, v2ray-geoip, v2ray-domain-list-community, assets ? [ v2ray-geoip v2ray-domain-list-community ]
+}:
+
+let
+  assetsDrv = symlinkJoin {
+    name = "v2ray-assets";
+    paths = assets;
+  };
+
+in
+buildGoModule rec {
+  pname = "v2ray";
+  version = "1.5.5";
+
+  src = fetchFromGitHub {
+    owner = "XTLS";
+    repo = "Xray-core";
+    rev = "v${version}";
+    sha256 = "x2aNAu+H3qJIKjQbE0rYxfQAWucvOMaU2eSy9YIZdcQ=";
+  };
+
+  vendorSha256 = "sha256-PBcUKUP06QKTu3djxoTb7B7Ld86fZz7+SDewRZy1YmQ=";
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  doCheck = false;
+
+  buildPhase = ''
+    buildFlagsArray=(-v -p $NIX_BUILD_CORES -ldflags="-s -w -buildid=")
+    runHook preBuild
+    go build "''${buildFlagsArray[@]}" -o xray ./main
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    install -Dm755 xray -t $out/bin
+
+    for file in $out/bin/*; do
+      makeWrapper "$file" "$out/bin/$(basename "$file")" \
+        --set-default XRAY_LOCATION_ASSET ${assetsDrv}/share/xray
+    done
+  '';
+
+  meta = {
+    description = "Xray, Penetrates Everything. Also the best v2ray-core, with XTLS support and fully compatible configuration";
```

Comment:
Could we provide a more useful description of what this software *does*?  I still don't know after reading this.  The current text sounds like some kind of tagline I'd find on an unhelpful blingy corporate website.

Reply:
The software description has been updated